tests/nativeapp/codegen/snowpark/test_extension_function_utils.py

```diff
@@ -7,6 +8,68 @@
 from tests.testing_utils.files_and_dirs import temp_local_dir
 
 
+def test_sanitize_ex_fn_attribute():
+    ex_fn = {"name": "john", "city": "toronto"}
+    ef_utils._sanitize_ex_fn_attribute(attr="name", ex_fn=ex_fn)  # noqa: SLF001
+    assert ex_fn["name"] == "john"
+    ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+        attr="name", ex_fn=ex_fn, make_uppercase=True
+    )
+    assert ex_fn["name"] == "JOHN"
+
+    ex_fn["city"] = ""
+    ef_utils._sanitize_ex_fn_attribute(attr="city", ex_fn=ex_fn)  # noqa: SLF001
+    assert ex_fn["city"] is None
+
+    ex_fn["team"] = "nade"
+    ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+        attr="team", ex_fn=ex_fn, make_uppercase=True, expected_type=str
+    )
+    assert ex_fn["team"] == "NADE"
+
+    optional_expected_type: Optional[List] = []
+    with pytest.raises(ef_utils.MalformedExtensionFunctionError):
+        ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+            attr="team",
+            ex_fn=ex_fn,
+            make_uppercase=True,
+            expected_type=type(optional_expected_type),
+        )
+
+    ex_fn["team"] = None
+    ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+        attr="team",
+        ex_fn=ex_fn,
+        make_uppercase=True,
+        expected_type=type(optional_expected_type),
+    )
+    assert ex_fn["team"] is None
+
+    ex_fn["team"] = ["nade"]
+    ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+        attr="team",
+        ex_fn=ex_fn,
+        make_uppercase=True,
+        expected_type=type(optional_expected_type),
+    )
+    assert ex_fn["team"] == ["nade"]
+
+    with pytest.raises(ef_utils.MalformedExtensionFunctionError):
+        ef_utils._sanitize_ex_fn_attribute(  # noqa: SLF001
+            attr="team", ex_fn=ex_fn, make_uppercase=True, expected_type=Dict
+        )
+
+
+def test_create_missing_attr_str():
```

Comment:
that's just an internal helper, IMHO this kind of stuff should be tested through other behaviour. It makes it so much harder to refactor code when we test the implementation instead of the contract.